pkg/queue/queue.go

```diff
@@ -166,19 +168,34 @@ func (q *Queue) GetConcurrency() int {
 	return q.concurrency
 }
 
-func (q *Queue) SetConcurrency(n int) {
+func (q *Queue) setConcurrency(n int) {
 	q.concurrencyMtx.Lock()
 	q.concurrency = n
 	q.concurrencyMtx.Unlock()
 }
 
+// setConcurrencyByPerReplica updates the total concurrency and per replica value
+func (q *Queue) setConcurrencyByPerReplica(newPerReplica uint32) {
+	q.concurrencyMtx.Lock()
+	defer q.concurrencyMtx.Unlock()
+
+	if q.concurrencyPerReplica == newPerReplica {
+		return
+	}
+
+	replicas := q.concurrency / int(q.concurrencyPerReplica)
```

Comment:
Shouldn't we get number of replicas from the system? The number of replicas fluctuates frequently and I think replicas counted this way would often not match the replicas in the actual system.

Maybe it should take a 2nd parameter: `replicas int`, because the deploy manager has the actual current replicas that kubernetes reports.